src/main/java/core/basesyntax/storage/Storage.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.storage;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitStorage = new HashMap<>();
+
+    public static Map<String, Integer> getFruitStorage() {
+        return Collections.unmodifiableMap(fruitStorage);
+    }
+
+    public static void addFruit(String fruit, int quantity) {
+        fruitStorage.merge(fruit, quantity, Integer::sum);
+    }
+
+    public static void setFruitQuantity(String fruit, int quantity) {
+        fruitStorage.put(fruit, quantity);
+    }
+
+    public static int getFruitQuantity(String fruit) {
+        return fruitStorage.getOrDefault(fruit, 0);
+    }
+}
```

Comment:
Consider adhering to the Single Responsibility Principle. While this `Storage` class is responsible for providing access to the fruit storage, the actual operations on the storage (addition, setting, getting) might be better encapsulated in a separate class that deals with storage manipulation. This would make the `Storage` class focused only on providing the storage without knowing how it's manipulated.